providers/implementations/signature/eddsa_sig.c

```diff
@@ -273,11 +429,20 @@ static int ed25519_digest_sign(void *vpeddsactx, unsigned char *sigret,
 #endif /* S390X_EC_ASM */
 
     if (peddsactx->prehash_flag) {
-        if (!EVP_Q_digest(peddsactx->libctx, SN_sha512, NULL, tbs, tbslen, md, &mdlen)
+        if (!peddsactx->prehash_by_caller_flag) {
+            if (!EVP_Q_digest(peddsactx->libctx, SN_sha512, NULL, tbs, tbslen, md, &mdlen)
                 || mdlen != EDDSA_PREHASH_OUTPUT_LEN)
+                return 0;
```

Comment:
Should there be an error pushed to the error stack in this case?

Reply:
Added